tools/mason/MasonBuild.chpl

```diff
@@ -19,74 +19,69 @@
  */
 
 
-private use List;
-private use Map;
-
-use TOML;
-use Spawn;
+use ArgumentParser;
 use FileSystem;
+use List;
+use Map;
 use MasonUtils;
 use MasonHelp;
 use MasonEnv;
 use MasonUpdate;
 use MasonSystem;
 use MasonExternal;
 use MasonExample;
+use Spawn;
+use TOML;
 
 proc masonBuild(args: [] string) throws {
-  var show = false;
-  var release = false;
-  var force = false;
+  const NOEXAMPLES = "**NO EXAMPLES**";
```

Comment:
This NOEXAMPLES part seems really hacky. Can you think of a better way to solve it? Can't we know there are no examples because there are 0 arguments in the list?

Reply:
> Can't we know there are no examples because there are 0 arguments in the list?

We can see that there are 0 arguments in the list, yes, but that doesn't tell us the difference between when the `--example` flag was given with no values vs when it was not given at all. The result of both cases is that there are 0 arguments in the list, which is why I had preloaded it with a known value. 

> ... seems really hacky

To address this feedback, I did it a slightly different way, but I don't like that I had to access the `_present` member variable. This use case may mean we need to expose that variable as part of the ArgumentParser API. For now, I have added some tests to the ArgumentParser to check for the expected behavior.

Reply:
I was supposing that `someArg.hasValue()` would do that but I am inferring that it returns `false` for an empty list? I had always thought of it as the same as `isPresent`.

Right now, the `hasValue` docs are this

```
    Check to see if a value was collected from the command line

    :returns:
              * `true` -- when at least one value was parsed
              * `false` -- when the argument wasn't provided and no
                default was defined
```

Does that imply that if you give it the default of the empty list, `hasValue` should only return false when the argument isn't provided? (I'm not entirely convinced the documentation is accurate here).

Anyway I think it's reasonable to improve on this area of the argument parser in future work & I'm happy with the `_present` workaround for now.

Reply:
Understandable that this would be a little confusing at first, but if the docs aren't making it clear I should update them.

If `hasValue()` returns true it means that either the user specified a value for the option, OR the user didn't specify a value, but we had a default that we used.

`hasValue()` is a conjunction of `_present` and `!values.isEmpty` and will return false if the argument wasn't provided and either there is no default value given or if the default was an empty list. Though we do set `_present` to true if we end up populating the value with the default, because the given default is an empty list, the resulting values list doesn't have any entries. Providing an empty list as the default is effectively the same as not providing a default at all, w.r.t `hasValue()`, although the behavior of `_present` would be different.

If a non-empty default value is given, then `hasValue()` will return true when the option is not present (populated with default), or if the option is present and a value was given (overrides the default), but will return false if the option is provided with no values.